src/socket_type.cpp

```diff
@@ -88,23 +85,17 @@ namespace aux {
 
 	void setup_ssl_hostname(socket_type& s, std::string const& hostname, error_code& ec)
 	{
-#if defined TORRENT_USE_OPENSSL
+#if defined TORRENT_USE_SSL
 #ifdef TORRENT_MACOS_DEPRECATED_LIBCRYPTO
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #endif
 		// for SSL connections, make sure to authenticate the hostname
 		// of the certificate
 #define CASE(t) case socket_type_int_impl<ssl_stream<t>>::value: \
-		s.get<ssl_stream<t>>()->set_verify_callback( \
-			boost::asio::ssl::rfc2818_verification(hostname), ec); \
-		ssl = s.get<ssl_stream<t>>()->native_handle(); \
-		ctx = SSL_get_SSL_CTX(ssl); \
+		s.get<ssl_stream<t>>()->set_host_name(hostname, ec); \
```

Comment:
I think it would be nice if the gnutls classes were compatible with boost.asio's ones as much as possible. iirc, asio actually has this verify callback with a generic signature, not dependent on OpenSSL

Reply:
The gnutls classes are compatible, for instance the signature is the same for `set_verify_callback`. The issue is that the `verify_context` is just a dummy container for an OpenSSL specific type.
However, I see now that I could add a `gnutls::rfc2818_verification` for this case.